### PR TITLE
feat: unit rent type

### PR DESCRIPTION
**Pull request by YazeedLoonat**
## Issue Overview

This PR addresses [#issue](https://github.com/bloom-housing/bloom/issues/3440)

- [x] This change addresses the issue in full

## Description
This creates all the endpoints for unit rent types

## How Can This Be Tested/Reviewed?
Follow the readme instructions for how to stand the new backend up
You should be able to test things out using the openai site at http://localhost:3101/api

you should be able to:
create, get a list of, update, get an individual, and delete unit rent types

permission is not implemented yet so don't worry about testing that yet

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have added QA notes to the issue with applicable URLs
- [x] I have performed a self-review of my own code
- [x] I have reviewed the changes in a desktop view
- [ ] I have reviewed the changes in a mobile view
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] I have assigned reviewers
- [x] I have run `yarn generate:client` and/or created a migration if I made backend changes that require them
- [x] My commit message(s) is/are polished, and any breaking changes are indicated in the message and are well-described
- [x] Commits made across packages purposefully have the same commit message/version change, else are separated into different commits

## Reviewer Notes:

Steps to review a PR:

- Read and understand the issue, and ensure the author has added QA notes
- Review the code itself from a style point of view
- Pull the changes down locally and test that the acceptance criteria is met
- Also review the acceptance criteria on the Netlify deploy preview (noting that these do not yet include any backend changes made in the PR)
- Either explicitly ask a clarifying question, request changes, or approve the PR if there are small remaining changes but the PR is otherwise good to go

## On Merge:

If you have one commit and message, squash. If you need each message to be applied, rebase and merge.

### CODE REVIEW

File: backend_new/src/services/unit-rent-type.service.ts [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+    this will return 1 unit rent type or error
+  */
+  async findOne(unitRentTypeId: string): Promise<UnitRentType> {
+    const rawUnitRentType = await this.prisma.unitRentTypes.findFirst({
+      where: {
+        id: {
+          equals: unitRentTypeId,
+        },
+      },
+    });
```
Comment (ludtkemorgan) [inline backend_new/src/services/unit-rent-type.service.ts:36]: Same here about the `findUnique`

Comment (ludtkemorgan) [reply backend_new/src/services/unit-rent-type.service.ts:36]: Actually, this findOne function is almost identical to the `findOrThrow` function below. Could you change the return of that function to the raw unit type and then this function calls that followed by the `mapTo` function?

File: backend_new/test/integration/unit-rent-type.e2e-spec.ts [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+    });
+
+    const res = await request(app.getHttpServer())
+      .get(`/unitRentTypes?`)
+      .expect(200);
+
+    expect(res.body.length).toEqual(2);
+    const sortedResults = res.body.sort((a, b) => (a.name < b.name ? 1 : -1));
+    expect(sortedResults[0].name).toEqual(unitRentTypeA.name);
+    expect(sortedResults[1].name).toEqual(unitRentTypeB.name);
```
Comment (ludtkemorgan) [inline backend_new/test/integration/unit-rent-type.e2e-spec.ts:41]: Same here as previous PRs

**Review by ludtkemorgan — APPROVED** [review]
🇺🇳

File: backend_new/src/services/unit-rent-type.service.ts (hunk 1 above)
Comment (YazeedLoonat) [reply backend_new/src/services/unit-rent-type.service.ts]: updated!

File: backend_new/test/integration/unit-rent-type.e2e-spec.ts (hunk 2 above)
Comment (YazeedLoonat) [reply backend_new/test/integration/unit-rent-type.e2e-spec.ts:41]: updated!

File: backend_new/types/src/backend-swagger.ts [hunk 3]
```diff
@@ -1158,3 +1314,18 @@ export enum UnitTypeEnum {
...
+  'visual' = 'visual',
+  'hearingAndVisual' = 'hearingAndVisual',
+  'mobilityAndVisual' = 'mobilityAndVisual',
+  'mobilityHearingAndVisual' = 'mobilityHearingAndVisual',
+}
+
+export enum UnitRentTypeEnum {
+  'fixed' = 'fixed',
+  'percentageOfIncome' = 'percentageOfIncome',
+}
```
Comment (seanmalbert) [inline backend_new/types/src/backend-swagger.ts:1331]: It looks like we weren't consistent with this with enums already, but I think enums should be ALL_CAPS or PascalCase/UpperCamelCase and are the single quotes necessary on the left side? Following the convention shown here: https://www.typescriptlang.org/docs/handbook/enums.html#string-enums (ignoring the all caps on the right side), UnitRentTypeEnum would be:
```
export enum UnitRentTypeEnum {
    Fixed = 'fixed'
    PercentageOfIncome = 'percentageOfIncome'
}

**Review by seanmalbert — APPROVED** [review]
@YazeedLoonat looks good. Please see my comment on the enums and resolve before merging.

File: backend_new/types/src/backend-swagger.ts (hunk 3 above)
Comment (YazeedLoonat) [reply backend_new/types/src/backend-swagger.ts:1331]: after offline convo with sean we're going to come to a consensus and address this in the standardization ticket 

https://www.prisma.io/docs/concepts/components/prisma-schema/data-model#defining-enums prisma doesn't allow for enums with a difference between its key and value so whatever we choose is what we end up choosing

**Pull request merged**